tests/integration/test_manylinux.py

```diff
@@ -232,526 +186,608 @@ def assert_show_output(manylinux_ctr, wheel, expected_tag, strict):
+                           '&& python -m pip wheel --no-deps -w /io .']
+        )
 
+        orig_wheel, *_ = os.listdir(io_folder)
+        assert orig_wheel.startswith("nonpy_rpath-0.1.0")
+        assert 'manylinux' not in orig_wheel
 
-@pytest.mark.parametrize(
-    'target_policy',
-    [f'{p}_{PLATFORM}' for p in list(MANYLINUX_IMAGES.keys())] +
-    [f'{p}_{PLATFORM}' for aliases in POLICY_ALIASES.values() for p in aliases]
-)
-@pytest.mark.parametrize('only_plat', [True, False])
-def test_build_wheel_compat(target_policy, only_plat, any_manylinux_container,
-                            docker_python, io_folder):
-    # test building wheels with compatibility with older spec
-    # check aliases for older spec
-
-    policy, tag, manylinux_ctr = any_manylinux_container
-
-    docker_exec(manylinux_ctr, [
-        'bash', '-c', 'cd /auditwheel_src/tests/integration/testsimple '
-                      '&& python -m pip wheel --no-deps -w /io .'
-    ])
-
-    filenames = os.listdir(io_folder)
-    orig_wheel = filenames[0]
-    assert 'manylinux' not in orig_wheel
-
-    if PLATFORM in {'x86_64', 'i686'}:
-        expect = f'manylinux_2_5_{PLATFORM}'
-        expect_tag = f'manylinux_2_5_{PLATFORM}.manylinux1_{PLATFORM}'
-    else:
-        expect = f'manylinux_2_17_{PLATFORM}'
-        expect_tag = f'manylinux_2_17_{PLATFORM}.manylinux2014_{PLATFORM}'
-
-    target_tag = target_policy
-    for pep600_policy, aliases in POLICY_ALIASES.items():
-        policy_ = f'{pep600_policy}_{PLATFORM}'
-        aliases_ = [f'{p}_{PLATFORM}' for p in aliases]
-        if target_policy == policy_ or target_policy in aliases_:
-            target_tag = f'{policy_}.{".".join(aliases_)}'
-
-    only_plat_arg = '--only-plat' if only_plat else ''
-    # we shall ba able to repair the wheel for all targets
-    docker_exec(manylinux_ctr, [
-        'bash', '-c',
-        f'auditwheel -v repair --plat {target_policy} {only_plat_arg} -w /io'
-        f' /io/{orig_wheel}'])
-    filenames = os.listdir(io_folder)
-    assert len(filenames) == 2
-    if only_plat or target_tag == expect_tag:
-        repaired_tag = target_tag
-    else:
-        repaired_tag = f'{expect_tag}.{target_tag}'
-    repaired_wheel = f'testsimple-0.0.1-{PYTHON_ABI}-{repaired_tag}.whl'
-    assert repaired_wheel in filenames
-    assert_show_output(manylinux_ctr, repaired_wheel, expect, True)
-
-    docker_exec(docker_python, f'pip install /io/{repaired_wheel}')
-    docker_exec(
-        docker_python,
-        ['python', '-c',
-         'from sys import exit; from testsimple import run; exit(run())']
-    )
+        # Repair the wheel using the appropriate manylinux container
+        repair_command = \
+            f'auditwheel repair --plat {policy} --only-plat -w /io /io/{orig_wheel}'
+        docker_exec(
+            manylinux_ctr,
+            ["bash", "-c", "LD_LIBRARY_PATH=/usr/local/lib " + repair_command]
```

Comment:
I had to add LD_LIBRARY_PATH for musllinux and I don't really know why yet. That's not needed on manylinux.

Reply:
What library was not found without LD_LIBRARY_PATH?

Reply:
That would be `libcrypt.so.2`

Reply:
I guess it's related to `parse_ld_so_conf` in `lddtree.py`. There's something to be parsed on manylinux but not on musllinux. There might be something else to be done on musl...

Reply:
The following code from musl seems relevant:
https://git.musl-libc.org/cgit/musl/tree/ldso/dynlink.c#n1078

Reply:
fixed in `lddtree.py` even though it should be cleaned up a bit there.